detox/src/realms/DetoxPrimaryContext.js

```diff
@@ -191,11 +194,18 @@ class DetoxPrimaryContext extends DetoxContext {
         await this[symbols.uninstallWorker]();
       }
     } finally {
-      if (this[_deviceAllocator]) {
-        await this[_deviceAllocator].cleanup();
-        this[_deviceAllocator] = null;
+      for (const key of Object.keys(this[_deviceAllocators])) {
+        const deviceAllocator = this[_deviceAllocators][key];
+        delete this[_deviceAllocators][key];
+        try {
+          await deviceAllocator.cleanup();
+        } catch (err) {
+          this[symbols.logger].error({ cat: 'device', err }, `Failed to cleanup device allocator for ${key}`);
+        }
       }
 
+      this[_cookieAllocators] = {};
```

Comment:
This begs for serious test coverage

Reply:
All the newly added rare `try-catch` branches have been covered.

The `cookieAllocators = {}` cleanup is covered by a new test, `DetoxPrimaryContext when initialized when a device is being allocated and then the context has been cleaned up should not be able to find that cookie anymore`.